docs/changelog/2754.bugfix.rst

```diff
@@ -0,0 +1,2 @@
+Setting ``[testenv] basepython = python3`` will no longer override the Python
```

Comment:
Wrap at 120 and add ```- by :user:`x`. ```

Reply:
> The following is intended to be a correct tox configuration:
> 
> ```ini
> [tox]
> min_version = 4.1
> env_list = py{38,39,310,311},docs
> 
> [testenv]
> base_python = python3.8
> 
> [testenv:docs]
> commands =
>   sphinx-build ...
> ```
> 
> The goal of this is to use `python3.8` (i.e. the value of `[testenv] base_python`) for all environments except those with specific `pyXX` factors in them. This helps eliminate issues where testenvs that do not specify a `pyXX` factor run with different Python versions in different environments.

Strong disagree. This is not a good way to express this. This config should hard fail for all envs expect py38.

Reply:
> But that makes `base_python` a special-case? For everything else, the idea is that `[testenv]` allows you to set "default" values that you can later override (or extend) in `[testenv:foo]` sections. tox itself does this for various things like `command`:

It's not a special case. Follows the same rule.
```ini
[testenv]
base_python = python3.8
[testenv:py39]
base_python = python3.9 
```

Would work. Where you're mistaken is that `py38` only sets `base_python` to `py38` if at no point is `base_python` specified. If you set it at `testenv` or `[testenv:py39]` level that no longer activates.